### ISSUE TITLE
Argument check in ImaUtil#updateAdDurationInAdGroup infrequently fails when using IMA extension

### ISSUE THREAD

**Issue by dbaelz** [google/ExoPlayer#10678]
### ExoPlayer Version

2.18.1

### Devices that reproduce the issue

The issue occurs infrequent on some devices based on Crashlytics logs. We were able to trigger it on two devices (marked with an asterisk). All other test devices played the same stream with the same ad insertion without any issues. Also, the crash doesn't happen on every ad insertion and right now we can't identify a pattern or reproduce it reliable.

### Devices
- Samsung Galaxy S21 5G (*)
- Pixel 2 Emulator with API 28 image (*)
- Samsung Galaxy S21s
- Samsung Galaxy Tab A7
- Samsung Galaxy Tab S4
- Xiaomi Redmi Note 8T
- Xiaomi Mi 11
- Xiaomi POCO X3 NFC
- HUAWEI P30
- HUAWEI MediaPad M5 lite
- HUAWEI P smart 2019
- Lenovo Tab M10 FHD Plus
- Lenovo Tab P10

### Devices that do not reproduce the issue

- Pixel 3
- Multiple device from different manufacturer like Samsung, Xiaomi and Huawei

### Reproducible in the demo app?

No

### Reproduction steps

- Implementation of the server-side ad insertion using the IMA extension as described in the [documentation](https://exoplayer.dev/ad-insertion.html#server-side-ad-insertion)
- Start a stream that supports DAI/ssai
- An ad should be displayed due DAI

Infrequent and without an identified pattern when showing an ad the [check in ImaUtil#updateAdDurationInAdGroup](https://github.com/google/ExoPlayer/blob/release-v2/extensions/ima/src/main/java/com/google/android/exoplayer2/ext/ima/ImaUtil.java#L326) fails and therefore the app crashes. We've checked the m3u8 file and our stream and can't find a reason for this behavior. Based on the stacktrace, we suspect that the Google IMA SDK or the calculation in the IMA extension could be the cause.

```
java.lang.IllegalArgumentException
        at com.google.android.exoplayer2.util.Assertions.checkArgument(Assertions.java:39)
        at com.google.android.exoplayer2.ext.ima.ImaUtil.updateAdDurationInAdGroup(ImaUtil.java:326)
        at com.google.android.exoplayer2.ext.ima.ImaServerSideAdInsertionMediaSource.addLiveAdBreak(ImaServerSideAdInsertionMediaSource.java:745)
        at com.google.android.exoplayer2.ext.ima.ImaServerSideAdInsertionMediaSource.access$2700(ImaServerSideAdInsertionMediaSource.java:109)
        at com.google.android.exoplayer2.ext.ima.ImaServerSideAdInsertionMediaSource$ComponentListener.onAdEvent(ImaServerSideAdInsertionMediaSource.java:882)
        at com.google.ads.interactivemedia.v3.internal.ahs.b(IMASDK:9)
        at com.google.ads.interactivemedia.v3.internal.ajh.b(IMASDK:12)
        at com.google.ads.interactivemedia.v3.internal.ain.k(IMASDK:47)
        at com.google.ads.interactivemedia.v3.internal.ait.f(IMASDK:5)
        at com.google.ads.interactivemedia.v3.internal.aiq.shouldOverrideUrlLoading(IMASDK:2)
        at android.webkit.WebViewClient.shouldOverrideUrlLoading(WebViewClient.java:77)
        at tY.a(PG:3)
        at xl.b(PG:81)
        at org.chromium.android_webview.AwContentsClientBridge.shouldOverrideUrlLoading(PG:172)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:326)
        at android.os.Looper.loop(Looper.java:160)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)        
```

### Expected result

DAI stream plays reliable without the failing `checkArgument` (and therefore crashing)

### Actual result

Infrequent and without an identified pattern when showing an ad the [check in ImaUtil#updateAdDurationInAdGroup](https://github.com/google/ExoPlayer/blob/release-v2/extensions/ima/src/main/java/com/google/android/exoplayer2/ext/ima/ImaUtil.java#L326) fails and therefore the app crashes. 

### Media

No reproducible and publicly available media/stream URL available for the issue.

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to dev.exoplayer@gmail.com after filing this issue.

**Comment by marcbaechinger** [comment]
Thanks for reporting. We are aware of this issue in case this is a live stream with prefetch enabled. For such stream it can happen that the number of ads in an adPod is increasing. Can you confirm this is the case for the stream with which you are seeing this?

**Comment by dbaelz** [comment]
Yes, I can confirm that. It's a live stream and in the Google Ad Manager the ad break setting `Prefetch ad breaks` (as mentioned [here](https://support.google.com/admanager/answer/7294289?hl=en)) is activated.

Is there a workaround or timeline for a fix for this issue?

**Comment by marcbaechinger** [comment]
I can't give a more concrete timeline than soon, but we will update this issue when a fix lands on dev-v2.

The only workaround for now is disable pre-fetch I think.

**Comment by marcbaechinger** [comment]
This is fixed with 2.18.2 (https://github.com/google/ExoPlayer/commit/f06e8ee736190d6b1951ddd3ed34de7593f52037)

I didn't tag the last commit correctly, so it does not show up here in the issue. Sorry for this. I close this issue. Please re-open if you you srtill see the issue.
